test/Microsoft.AspNetCore.Mvc.Razor.Test/RazorPageTest.cs

```diff
@@ -497,6 +514,28 @@ public void IsSectionDefined_ThrowsIfPreviousSectionWritersIsNotRegistered()
         }
 
         [Fact]
+        public async Task EnsureRenderedBodyOrSections_SucceedsIfDefinedSectionsAreNotRendered_AndIgnored()
```

Comment:
In this test, you'd do something along the lines of 

```C#
page.PreviousSectionWriters = new Dictionary<string, RenderAsyncDelegate>
{
           { "section1", _nullRenderAsyncDelegate },
           { "section2", _nullRenderAsyncDelegate },
};

page.IgnoreSection("section1");

// Assert section2 is rendered.
```
Just to make sure we don't do something bad and ignore all sections in general after a `IgnoreSection` is called.